PWGHF/TableProducer/candidateCreatorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,655 @@
+        registry.fill(HIST("hCovPVXX"), covMatrixPV[0]);
+        registry.fill(HIST("hCovPVYY"), covMatrixPV[2]);
+        registry.fill(HIST("hCovPVXZ"), covMatrixPV[3]);
+        registry.fill(HIST("hCovPVZZ"), covMatrixPV[5]);
+        // covariance matrix elements of SV
+        registry.fill(HIST("hCovSVXX"), covMatrixSV[0]);
+        registry.fill(HIST("hCovSVYY"), covMatrixSV[2]);
+        registry.fill(HIST("hCovSVXZ"), covMatrixSV[3]);
+        registry.fill(HIST("hCovSVZZ"), covMatrixSV[5]);
+        // DCAs of prongs
+        registry.fill(HIST("hDcaXYProngs"), trackCasc.getPt(), impactParameterCasc.getY());
+        registry.fill(HIST("hDcaXYProngs"), trackCharmBachelor0.pt(), impactParameter0.getY());
+        registry.fill(HIST("hDcaXYProngs"), trackCharmBachelor1.pt(), impactParameter1.getY());
+        registry.fill(HIST("hDcaZProngs"), trackCasc.getPt(), impactParameterCasc.getZ());
+        registry.fill(HIST("hDcaZProngs"), trackCharmBachelor0.pt(), impactParameter0.getZ());
+        registry.fill(HIST("hDcaZProngs"), trackCharmBachelor1.pt(), impactParameter1.getZ());
+      }
+
+      //---------------------------------fill candidate table rows-------------------------------------------------------------------------------------------
+      rowCandidateBase(collision.globalIndex(),
+                       primaryVertex.getX(), primaryVertex.getY(), primaryVertex.getZ(),
+                       covMatrixPV[0], covMatrixPV[2], covMatrixPV[5],
+                       /*3-prong specific columns*/
+                       rowTrackIndexXicPlus.cascadeId(), rowTrackIndexXicPlus.prong0Id(), rowTrackIndexXicPlus.prong1Id(),
+                       casc.bachelorId(), casc.posTrackId(), casc.negTrackId(),
+                       secondaryVertex[0], secondaryVertex[1], secondaryVertex[2],
+                       covMatrixSV[0], covMatrixSV[2], covMatrixSV[5],
+                       errorDecayLength, errorDecayLengthXY,
+                       chi2SV, massXiPiPi, signXic,
+                       pVecXi[0], pVecXi[1], pVecXi[2],
+                       pVecPi0[0], pVecPi0[1], pVecPi0[2],
+                       pVecPi1[0], pVecPi1[1], pVecPi1[2],
+                       impactParameterCasc.getY(), impactParameter0.getY(), impactParameter1.getY(),
+                       std::sqrt(impactParameterCasc.getSigmaY2()), std::sqrt(impactParameter0.getSigmaY2()), std::sqrt(impactParameter1.getSigmaY2()),
+                       /*cascade specific columns*/
+                       vertexCasc[0], vertexCasc[1], vertexCasc[2],
+                       vertexV0[0], vertexV0[1], vertexV0[2],
+                       cpaXi, cpaXYXi, cpaLambda, cpaXYLambda,
+                       massXiPi0, massXiPi1);
+    } // loop over track triplets
+  }
+  PROCESS_SWITCH(HfCandidateCreatorXicToXiPiPi, processXicPlusWithDcaFitter, "Run candidate creator with DCAFitter.", true);
+
+  void processXicPlusWithKFParticleFromDerivedData(aod::Collisions const&,
```

Comment:
Why do you need the suffix `FromDerivedData`? There is not alternative without it.

Reply:
I had another process function initially which did the track-index skimming and then built the candidates directly. I dediced not to remove the suffix to make clear that this process function (or more precisely, the KF version of the workflow) works only when one runs on derived data, otherwise the trackIndexSkimCreatorLfCascades will run over empty tables (CascCores+ CascCovs which are not filled before, instead of KFCascCores+KFCascCovs) and no output will be produced.
If you want, I can remove it.

Reply:
Yes please